contrib/ruby/ext/trilogy-ruby/cext.c

```diff
@@ -933,6 +936,32 @@ static VALUE rb_trilogy_closed(VALUE self)
     }
 }
 
+static VALUE rb_trilogy_discard(VALUE self)
+{
+    struct trilogy_ctx *ctx = get_ctx(self);
+    if (ctx->conn.socket == NULL) {
+        return Qnil;
+    }
+
+    int sock_fd = trilogy_sock_fd(ctx->conn.socket);
+    if (sock_fd >= 0) {
+        int null_fd = open("/dev/null", O_RDWR | O_CLOEXEC);
+        if (null_fd < 0) {
+            trilogy_syserr_fail_str(errno, rb_str_new_cstr("Failed to open /dev/null"));
+            return Qfalse;
+        }
+
+        if (dup2(null_fd, sock_fd) < 0) {
+            trilogy_syserr_fail_str(errno, rb_str_new_cstr("dup2 failed"));
+        }
+        close(null_fd);
+    }
```

Comment:
I don't know if it's better to have all this logic here, or it it would make sense to implement a `trilogy_discard` in the C library that could be used by other potential bindings?

Reply:
IMHO it would make sense to add a method to the C API that handles duping the `/dev/null`  file descriptor to the socket fd, and returns either `TRILOGY_OK` or `TRILOGY_SYSERR`, and then the cext could handle raising the ruby error.